### PR TITLE
SISRP-23117 Guard against overlarge Oracle "SELECT IN ()" queries

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.berkeley.edu/browse/SISRP-23117

### CODE REVIEW

File: app/models/user/basic_attributes.rb [hunk 1]
```diff
@@ -5,9 +5,12 @@ module BasicAttributes
     def attributes_for_uids(uids)
       return [] if uids.blank?
       uid_set = uids.to_set
-      attrs = CampusOracle::Queries.get_basic_people_attributes(uids).map do |result|
-        uid_set.delete result['ldap_uid']
-        transform_campus_row result
+      # Oracle dislikes ' IN ()' queries with more than 1000 items.
+      while (next_batch = uids.slice!(0..999)).present?
```
Comment (paulkerschen) [inline app/models/user/basic_attributes.rb]: Oh, for heaven's sake, I thought we'd caught all of those. Thanks for bagging this one, @raydavis. What do you think about the idiomatic writing `uids.each_slice(1000) do |next_batch|`?

File: app/models/user/basic_attributes.rb [hunk 2]
```diff
@@ -5,9 +5,12 @@ module BasicAttributes
...
+        attrs = CampusOracle::Queries.get_basic_people_attributes(next_batch).map do |result|
```
Comment (paulkerschen) [inline app/models/user/basic_attributes.rb]: @raydavis Also, perhaps I'm reading wrong, but doesn't the assignment of `attrs` within the loop mean that you end up with only the results of the last iteration? (Submit 1003 UIDs, get 3 results.)

File: app/models/user/basic_attributes.rb (hunk 1 above)
Comment (raydavis) [inline app/models/user/basic_attributes.rb]: I'd think I missed it. Thanks, @paulkerschen , I'll amend.

File: app/models/user/basic_attributes.rb (hunk 2 above)
Comment (raydavis) [inline app/models/user/basic_attributes.rb]: Ouch. I'll _really_ amend.

**Comment by raydavis** [comment]
Peace on earth, good will toward men, amended.

**Pull request merged**